internal/resources/aws/util.go

```diff
@@ -682,33 +687,45 @@ var reservedPaymentOptionMapping = map[string]string{
 	"all_upfront":     "All Upfront",
 }
 
-type reservedInstanceResolver struct {
-	term          string
-	paymentOption string
-	dbInstance    bool
+var elasticacheReservedNodeCacheLegacyOfferings = map[string]string{
+	"heavy_utilization":  "Heavy Utilization",
+	"medium_utilization": "Medium Utilization",
+	"light_utilization":  "Light Utilization",
 }
 
-func (r reservedInstanceResolver) Term() string {
-	return reservedTermsMapping[r.term]
+var elasticacheReservedNodeLegacyTypes = []string{"t2", "m3", "m4", "r3", "r4"}
+
+// RDS implementation of reservationResolver
+type rdsReservationResolver struct {
+	term          string
+	paymentOption string
 }
 
-func (r reservedInstanceResolver) PaymentOption() string {
-	return reservedPaymentOptionMapping[r.paymentOption]
+func newRdsReservationResolver(term, paymentOption string) reservationResolver {
+	return &rdsReservationResolver{
+		term:          term,
+		paymentOption: paymentOption,
+	}
 }
```

Comment:
see other comments about godoc and constructors

Reply:
Removed constructor